express-notes-api/index.js

```diff
@@ -0,0 +1,160 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+
+app.get('/api/notes', (req, res) => {
+  const gradesArray = [];
+  fs.readFile('data.json', 'utf8', (err, data) => {
+    data = JSON.parse(data);
+    const objArray = [];
+    for (const prop in data.notes) {
+      objArray.push({
+        "content": data.notes[prop].content,
+        "id": data.notes[prop].id
+      })
+    }
+    res.json(objArray)
+  })
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  if(req.params.id <= 0) {
+    console.log('gang gang')
+    res.status(400)
+    res.json({"error": "id must be a positive integer"})
+  } else {
+    fs.readFile('data.json', 'utf8', (err, data) => {
+      if(err) throw err;
+      data = JSON.parse(data);
+      const objArray = [];
+      for (const prop in data.notes) {
+        objArray.push({
+          "content": data.notes[prop].content,
+          "id": data.notes[prop].id
+        })
+      };
+      var placement;
+      for (var i = 0; objArray.length > i; i++) {
+        if (objArray[i].id === parseInt(req.params.id)) {
+          placement = i;
+          break;
+        }
+      }
+      if(placement === undefined) {
+        res.status(404);
+        res.json({"error": `cannot find note with id ${req.params.id}`});
+      } else {
+        res.json(objArray[placement]);
+      }
+    });
+  }
+});
+
+app.use(express.json())
+
+app.post('/api/notes', (req, res) => {
+  if (req.body.content === undefined) {
+    res.status(400);
+    res.json({
+      "error": "content is a required field"
+    })
+  } else {
+    fs.readFile('data.json', 'utf8', (err, data) => {
+      data = JSON.parse(data)
+      const write = {
+        "content": req.body.content,
+        "id": data.nextId
+      }
+      let nextId = data.nextId
+      data.notes[nextId] = write;
+      data.nextId++;
+      data = JSON.stringify(data);
+      fs.writeFile('data.json', data, (err) => {
+        if(err) {
+          res.status(500);
+          res.json({
+            "error": "An unexpected error occurred."
+          })
+        } else {
+          res.status(201);
+          res.json(write);
+        }
+      })
+    })
+  }
+});
+
+app.delete('/api/notes/:id', (req, res) => {
+  if(req.params.id <= 0) {
+    res.status(400);
+    res.json({
+      "error": "id must be a positive integer"
+    })
```

Comment:
Same as above: this doesn't catch `id`s that aren't numbers (ex: `'trololololo'`) or are fractional (`ex: `42.34`).